app/controllers/user_panel/orders_controller.rb

```diff
@@ -0,0 +1,42 @@
+module UserPanel
+  class OrdersController < ApplicationController
+    before_action :authenticate_user!
+
+    def show
+      attach_order_to_user
+    end
+
+    def index
+      @orders = current_user.orders
+    end
+
+    def confirm
+      @order = current_user.orders.pending.find(params[:id])
+      @order.update status: Order::COMPLETE_STATUS
+      redirect_to user_panel_order_path(@order)
+    end
+
+    private
+
+    def attach_order_to_user
+      @order = current_user.orders.find_by(id: params[:id])
+      return @order if @order.present?
+
+      @order = current_guest.orders.find_by(id: params[:id])
+
+      raise ActiveRecord::RecordNotFound unless @order.present?
+
+      do_attachment
+    end
+
+    def do_attachment
+      if Event.decrease_ticket_number(@order)
```

Comment:
Trochę pachnie mi to programowaniem strukturalnym opakowanym w modele. Tzn. klasa `Event` musi wiedzieć wszystko o tym jak działają zamówienia, żeby móc zrealizować tę metodę. Zostawię tu propozycję tego jak ja bym to zrobił, ale nie trzeba się do tego stosować. Głównie chodzi o to, że metody mają tylko jedną odpowiedzialność i nie wiedzą za dużo o całym procesie.

```ruby
class Order

def items_available?
  order_item.all? { |item| item.available? } # przydatna metoda https://ruby-doc.org/core-2.5.1/Enumerable.html#method-i-all-3F
end

def decrease_tickets_amount!
  order_item.each do |item|
    item.decrease_amount!
end
```

```ruby
class OrderItem

def available?
  (event.amount - amount) >= 0
end

def decrease_amount!
  event.decrease_amount_by!(amount)
end
```

```ruby
class Event

def decrease_amount_by!(number)
  update!(amount: amount - number)
end
```

```ruby
def do_attachment
  if @order.items_available?
    @order.decrease_tickets_amount!
    @order.update owner: current_user
  else
     # ta część tak smao jak u Pana
```